src/appleseed/renderer/modeling/frame/frame.cpp

```diff
@@ -680,6 +719,116 @@ void Frame::extract_parameters()
         Vector2u(0, 0),
         Vector2u(impl->m_frame_width - 1, impl->m_frame_height - 1));
     impl->m_crop_window = m_params.get_optional<AABB2u>("crop_window", default_crop_window);
+
+    // Retrieve the denoiser parameters.
+    {
+        string denoise_mode = m_params.get_optional<string>("denoiser", "off");
+
+        if (denoise_mode == "off")
+            impl->m_denoise_mode = DenoiseMode::Off;
+        else if (denoise_mode == "on")
+            impl->m_denoise_mode = DenoiseMode::Denoise;
+        else if (denoise_mode == "write_outputs")
+            impl->m_denoise_mode = DenoiseMode::WriteOutputs;
+        else
+        {
+            RENDERER_LOG_ERROR(
+                "invalid value \"%s\" for parameter \"%s\", using default value \"%s\".",
+                denoise_mode.c_str(),
+                "denoiser",
+                "off");
+            impl->m_denoise_mode = DenoiseMode::Off;
+        }
+
+        // Override the reconstruction filter parameter if the denoiser is not off
```

Comment:
Wow, really. Why do we need to do that? What happens if we don't?

Reply:
We compute the histograms and covariances using a box filter with a 0.5 radius.
By overriding the filter, we do the same for the beauty and AOV images.

I didn't test enough to know if it makes a real difference. It's something we need to check.